packages/textfield/src/mwc-textfield-base.ts

```diff
@@ -221,26 +229,25 @@ export abstract class TextFieldBase extends FormElement {
         ${this.iconTrailing ? this.renderIcon(this.iconTrailing) : ''}
         ${this.outlined ? this.renderOutlined() : this.renderLabelText()}
       </div>
-      ${
-        (this.helper || this.validationMessage || this.charCounter) ?
-            this.renderHelperText() :
-            ''}
+      ${this.renderHelperText(this.renderCharCounter())}
     `;
   }
 
   updated(changedProperties: PropertyValues) {
-    const charCounter =
-        changedProperties.get('charCounter') as boolean | undefined;
-
-    // update foundation only when charCounter goes from false to true
-    if (!charCounter && this.charCounter) {
+    const maxLength = changedProperties.get('maxLength') as number | undefined;
+
+    const mLChangedFromUndef = maxLength === -1 && this.maxLength !== -1;
+    const mLChangedToUndef =
+        maxLength !== undefined && maxLength !== -1 && this.maxLength === -1;
+    // update foundation only when maxLength goes from undefined (-1) to true
```

Comment:
Can you replace this with a comment explaining *why* we update (or is it re-create?) the foundation in this case (rather than describing *what* the code does)?